src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -1390,11 +1390,12 @@ internal void WriteStringToListener(EventListener listener, string msg, SessionM
                 }
                 else
                 {
-                    List<object> arg = new List<object>();
-                    arg.Add(msg);
                     EventWrittenEventArgs eventCallbackArgs = new EventWrittenEventArgs(this);
                     eventCallbackArgs.EventId = 0;
-                    eventCallbackArgs.Payload = new ReadOnlyCollection<object>(arg);
+                    eventCallbackArgs.Message = msg;
+                    eventCallbackArgs.Payload = new ReadOnlyCollection<object>(new List<object>() { msg });
+                    eventCallbackArgs.PayloadNames = new ReadOnlyCollection<string>(new List<string> { "message" });
```

Comment:
Yes, we could use arrays, and it would probably be better (only probably because the actual access to the elements is actually slower when you access arrays through the IList<T> interface).   But this is just not an interesting perf path (it is only called on error/warning conditions), and the goal of this update is to fix a bug, not improve perf.

If you wish it is reasonable to submit a pull request for the change you want, but I don't think it is worth it...